### PR TITLE
[1LP][RFR] Automating Ansible tests for Branched Repo and Link to playbook

### PR DESCRIPTION
**Pull request by sbulage**
Already automated manual tests: `test_embed_tower_repo_list`, `test_embed_tower_repos_available`.
This PR automating manual tests: `test_embed_tower_repo_links`, `test_embed_tower_add_branch_repo`, `test_embed_tower_playbook_links`.

PRT Run:

{{ pytest: cfme/tests/ansible/test_embedded_ansible_basic.py -k "test_embedded_ansible_repository_playbook_link or test_embedded_ansible_repository_branch_crud" -svvvv --long-running }}

### CODE REVIEW

File: cfme/ansible/repositories.py [hunk 1]
```diff
@@ -75,6 +75,16 @@ def is_displayed(self):
         )
 
 
+class PlaybookRepositoryView(RepositoryDetailsView):
```
Comment (digitronik) [inline cfme/ansible/repositories.py]: Optional:
I think the name should be `RepositoryPlaybooksView`.... as its playbooks of specific repositories

File: cfme/tests/ansible/test_embedded_ansible_basic.py [hunk 2]
```diff
@@ -131,6 +133,35 @@ def test_embedded_ansible_repository_crud(ansible_repository, wait_for_ansible):
...
+        assignee: sbulage
+        casecomponent: Ansible
+        caseimportance: critical
+        initialEstimate: 1/12h
+        tags: ansible_embed
+    """
+    repositories = appliance.collections.ansible_repositories
+    repository = repositories.create(
+        name=fauxfactory.gen_alpha(),
+        url="https://github.com/ManageIQ/integration_tests_playbooks.git",
```
Comment (digitronik) [inline cfme/tests/ansible/test_embedded_ansible_basic.py]: I think better to pick this URL from yamls not hardcore here... `ansible_links.embedded_ansible`

File: cfme/tests/ansible/test_embedded_ansible_basic.py [hunk 3]
```diff
@@ -131,6 +133,35 @@ def test_embedded_ansible_repository_crud(ansible_repository, wait_for_ansible):
     assert view.description.value == updated_description
 
 
+@pytest.mark.rhel_testing
+@pytest.mark.tier(1)
+def test_embedded_ansible_repository_branch_crud(appliance, wait_for_ansible):
```
Comment (digitronik) [inline cfme/tests/ansible/test_embedded_ansible_basic.py]: Q:
As its crud test. it's possible to add update some parameters?

File: cfme/tests/ansible/test_embedded_ansible_basic.py [hunk 4]
```diff
@@ -131,6 +133,35 @@ def test_embedded_ansible_repository_crud(ansible_repository, wait_for_ansible):
...
+        description=fauxfactory.gen_alpha(),
+        scm_branch="second_playbook_branch",
+    )
+    view = navigate_to(repository, "Details")
+    assert (
+        view.entities.summary("Repository Options").get_text_of("SCM Branch")
+        == repository.scm_branch
+    )
+
+    repository.delete()
```
Comment (digitronik) [inline cfme/tests/ansible/test_embedded_ansible_basic.py]: I think we need to check `.exists` after creation and `not .exists` after deletion of a repository as test pointing to crude.

File: cfme/tests/ansible/test_embedded_ansible_basic.py (hunk 3 above)
Comment (sbulage) [reply cfme/tests/ansible/test_embedded_ansible_basic.py]: Updating name's and all it is already covered in the Repository_CRUD tests.

File: cfme/tests/ansible/test_embedded_ansible_basic.py (hunk 4 above)
Comment (sbulage) [reply cfme/tests/ansible/test_embedded_ansible_basic.py]: Above `assert` statement and `view` covers the `.exists` of repository.

File: cfme/tests/ansible/test_embedded_ansible_basic.py (hunk 2 above)
Comment (sbulage) [reply cfme/tests/ansible/test_embedded_ansible_basic.py]: Yup. I will update it.

File: cfme/ansible/repositories.py [hunk 5]
```diff
@@ -75,6 +75,16 @@ def is_displayed(self):
         )
 
 
+class RepositoryPlaybookView(RepositoryDetailsView):
```
Comment (izapolsk) [inline cfme/ansible/repositories.py]: does this view show one playbook or many ?

File: cfme/tests/ansible/test_embedded_ansible_basic.py [hunk 6]
```diff
@@ -131,6 +134,43 @@ def test_embedded_ansible_repository_crud(ansible_repository, wait_for_ansible):
...
+        repository = repositories.create(
+            name=fauxfactory.gen_alpha(),
+            url=getattr(playbooks_yaml, playbook_name),
+            description=fauxfactory.gen_alpha(),
+            scm_branch="second_playbook_branch"
+        )
+    except (KeyError, AttributeError):
+        message = "Missing ansible_links content in cfme_data, cannot setup repository"
+        logger.exception(message)  # log the exception for debug of the missing content
+        pytest.skip(message)
```
Comment (izapolsk) [inline cfme/tests/ansible/test_embedded_ansible_basic.py]: why skip not fail ?

File: cfme/tests/ansible/test_embedded_ansible_basic.py [hunk 7]
```diff
@@ -131,6 +134,43 @@ def test_embedded_ansible_repository_crud(ansible_repository, wait_for_ansible):
...
+    view = navigate_to(repository, "Details")
+    assert (
+        view.entities.summary("Repository Options").get_text_of("SCM Branch")
+        == repository.scm_branch
+    )
```
Comment (izapolsk) [inline cfme/tests/ansible/test_embedded_ansible_basic.py]: I'd ask you to add appropriate assert message and you can easily make this statement more readable by assiging view field value to variable 

`scm_branch = view.entities.summary("Repository Options").get_text_of("SCM Branch")`

File: cfme/tests/ansible/test_embedded_ansible_basic.py (hunk 4 above)
Comment (izapolsk) [reply cfme/tests/ansible/test_embedded_ansible_basic.py]: it doesn't fully cover this case. if your test fails on assert, your repository won't be deleted.

File: cfme/ansible/repositories.py (hunk 5 above)
Comment (sbulage) [reply cfme/ansible/repositories.py]: It shows All Playbooks, when navigated.

File: cfme/tests/ansible/test_embedded_ansible_basic.py (hunk 7 above)
Comment (sbulage) [reply cfme/tests/ansible/test_embedded_ansible_basic.py]: Okay I will update.

File: cfme/tests/ansible/test_embedded_ansible_basic.py (hunk 4 above)
Comment (sbulage) [reply cfme/tests/ansible/test_embedded_ansible_basic.py]: I will add `repository.delete_if_exists()` in finalizer.

File: cfme/tests/ansible/test_embedded_ansible_basic.py (hunk 6 above)
Comment (sbulage) [reply cfme/tests/ansible/test_embedded_ansible_basic.py]: Updated with fail.

File: cfme/ansible/repositories.py (hunk 5 above)
Comment (izapolsk) [reply cfme/ansible/repositories.py]: So, I guess it should be called RepositoryPlaybooksView, right ?

**Comment by sbulage** [comment]
CFME 5.11 failed due to Sprout failed to give appliance.

File: cfme/ansible/repositories.py (hunk 5 above)
Comment (sbulage) [reply cfme/ansible/repositories.py]: @izapolsk Updated it.

File: cfme/tests/ansible/test_embedded_ansible_basic.py [hunk 8]
```diff
@@ -131,6 +134,44 @@ def test_embedded_ansible_repository_crud(ansible_repository, wait_for_ansible):
...
+        initialEstimate: 1/12h
+        tags: ansible_embed
+    """
+    repositories = appliance.collections.ansible_repositories
+    try:
+        playbooks_yaml = cfme_data.ansible_links.playbook_repositories
+        playbook_name = getattr(request, 'param', 'embedded_ansible')
+        repository = repositories.create(
+            name=fauxfactory.gen_alpha(),
+            url=getattr(playbooks_yaml, playbook_name),
```
Comment (izapolsk) [inline cfme/tests/ansible/test_embedded_ansible_basic.py]: what is intention of getattr here ? I guess it should work w/o it as well

Comment (sbulage) [reply cfme/tests/ansible/test_embedded_ansible_basic.py]: Earlier I have hard-coded the Repository URL, then @digitronik suggested to get it from yamls.

**Pull request merged**